sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1615 @@
+
+Available only if `propertyListA` and `propertyListB` are specializations of the `properties` class.
+
+a@
+[source]
+----
+template <typename T,
+          typename propertyListA,
+          typename propertyListB>
+annotated_ptr<T, propertyListB>
+annotated_malloc(
+   size_t count,
+   const queue& syclQueue,
+   const propertyListA &propList)
+----
+a@ `propList` must contain the `sycl::ext::oneapi::experimental::usm_kind` property, otherwise the compiler will issue a diagnostic error.
+Returns a USM memory allocation of the kind specified by the `sycl::ext::oneapi::experimental::usm_kind` property.
+This allocation is specified in number of elements of type `T`.
+This memory must be deallocated with `sycl::free` to avoid memory leaks.
+On failure, the raw pointer of the returned `annotated_ptr` will be `nullptr`.
+
+Additional runtime and compile-time constant properties can be provided in `propList`.
+The compile-time constant properties in `propList` will also appear as properties of the returned `annotated_ptr`.
+Runtime properties in `propList` will not appear as properties of the returned `annotated_ptr`.
+
+Simplified form where `syclQueue` provides the `context`
+and any necessary `device`.
+
+Available only if `propertyListA` and `propertyListB` are specializations of the `properties` class.
+
+
+|====
+
+=== USM Memory Allocation Properties
+
+This section specifies the properties that can be passed to the USM memory allocation functions with `properties` support defined in Section <<section.usm.allocs>>.
+A SYCL implementation or SYCL backend may support additional properties other than those defined here, provided they are defined in accordance with the
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties] extension.
+If unsupported properties are passed to the USM memory allocation functions with `properties` support, the compiler will issue a diagnostic error.
+
+All properties supported by `annotated_ptr` are supported by the USM memory allocation functions with `properties` support. These properties do not affect the behavior of the allocation functions at runtime, except where explicitly specified.
+
+
+[NOTE]
+====
+The USM memory allocation functions with `properties` support defined in this extension support all properties supported by `annotated_ptr`, as a convenience. This allows the user to attach all desired properties to the `annotated_ptr` in a single step.
+====
+
+Table <<table.usm.malloc.properties>> lists properties that affect the behavior of the allocation functions.
+
+[[table.usm.malloc.properties]]
+.Properties that affect the behavior of the USM memory allocation functions with properties support
+[options="header"]
+|====
+|Property|Description|Source
+a|
+sycl::ext::oneapi::experimental::alignment
+| 	
+If this property is passed to a USM memory allocation function with `properties` support, it instructs the runtime to allocate memory with this alignment in bytes.
```

Comment:
I think we should add a statement like the following to be consistent with the existing allocation functions that take an alignment and to be consistent with C++ `std::aligned_alloc`:

> The set of allowed alignments is implementation defined.  Specifying an alignment that is not supported causes the allocation function to return a null `annotated_ptr`.

Reply:
Ok, I added these statements to the table.
Note that the allocation functions would return a non-null annotated_ptr object, whose raw pointer is null.